plugins/chain_plugin/include/eosio/chain_plugin/account_query_db.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+#include <eosio/chain/types.hpp>
+#include <eosio/chain/block_state.hpp>
+#include <eosio/chain/trace.hpp>
+
+namespace eosio::chain_apis {
+   /**
+    * This class manages the ephemeral indices and data that provide the `get_accounts_by_authorizers` RPC call
+    * There is no persistence and the indices/caches are recreated when the class is instantiated based on the
+    * current state of the chain.
+    */
+   class account_query_db {
+   public:
+
+      /**
+       * Instantiate a new account query DB from the given chain controller
+       * The caller is expected to manage lifetimes such that this controller reference does not go stale
+       * for the life of the account query DB
+       * @param chain - controller to read data from
+       */
+      account_query_db( const class eosio::chain::controller& chain );
+      ~account_query_db();
+
+      /**
+       * Allow moving the account query DB (including by assignment)
+       */
+      account_query_db(account_query_db&&);
```

Comment:
Not implemented. Also I doubt the operator= is used, I think you should remove them.